units/systemd-machine-id-commit.service

```diff
@@ -12,8 +12,8 @@ Description=Commit a transient machine-id on disk
 Documentation=man:systemd-machine-id-commit.service(8)
 DefaultDependencies=no
 Conflicts=shutdown.target
-Before=sysinit.target shutdown.target
-After=local-fs.target
+Before=shutdown.target
+After=basic.target
```

Comment:
So I can see why dropping the before= line makes sense. But I am not convinced about the After= change. I think we should still commit the uuid as early as possible, and not needlessly delay that. I see no reason to say that basic.target was a requirement to say that first boot was complete now. I mean, I think in many cases people might very well declare that running the preset stuff during early boot fully would be enough to not require us to re-run first boot mode. Hence running the service as early as possible should be OK.

Or to say this differently: I think OS builders should think about where to put the boundary for this, and explicitly order all services they want to finish fist before this boundary. I see no reason to do that for them.

By dropping the Before= line people can order both early-boot services before and regular services before this commit point, which makes a lot of sense. But other than that I'd not make a change here.

Reply:
Okay, I was under the assumption that `m->first_boot` will be reset once basic.target is reached but upon reading the code again, it seems this happens after all jobs of the initial transaction completed?

What you're saying makes a lot of sense, I agree that committing should happen as early as possible and allowing regular services to be a part of the first boot is probably a nice feature as well.  Though, maybe a `first-boot.target` for the purposes of ordering makes more sense then?

Reply:
yes, i came to the same conclusion that f-b.t would make sense. see below.